dataikuapi/dss/ml.py

```diff
@@ -308,50 +332,679 @@ def set_metric(self, metric=None, custom_metric=None, custom_metric_greater_is_b
+        self._raw_settings["distributed"] = distributed
+
+    def set_n_jobs(self, n_jobs):
+        assert isinstance(n_jobs, int) and n_jobs > 0
+        self._raw_settings["nJobs"] = n_jobs
+
+
+class HyperparameterSettings(object):
+
+    def __init__(self, name, algo_settings):
+        self.name = name
+        self._algo_settings = algo_settings
+
+
+class NumericalHyperparameterSettings(HyperparameterSettings):
+
+    def __repr__(self):
+        clean_dict = self._algo_settings._get_active_numerical_hyperparam_dict(self.name)
+        return self.__class__.__name__ + "(hyperparameter=\"{}\", settings={})".format(self.name, json.dumps(clean_dict, indent=4))
+
+    __str__ = __repr__
+
+    def set_explicit_values(self, values, mode_as_explicit=True):
+        self._algo_settings._set_numerical_explicit_values(self.name, values, mode_as_explicit=mode_as_explicit)
+
+    def set_range(self, range_min=None, range_max=None, nb_values=None, mode_as_range=True):
+        self._algo_settings._set_numerical_range(self.name, range_min=range_min, range_max=range_max, nb_values=nb_values,
+                                                 mode_as_range=mode_as_range)
+
+    def set_mode_as_explicit(self):
+        self._algo_settings._set_hyperparameter_mode_as_explicit(self.name)
+
+    def set_mode_as_range(self):
+        self._algo_settings._set_hyperparameter_mode_as_range(self.name)
+
+
+class CategoricalHyperparameterSettings(HyperparameterSettings):
+
+    def __repr__(self):
+        return self.__class__.__name__ + "(hyperparameter=\"{}\", settings={})".format(self.name, json.dumps(self._algo_settings[self.name], indent=4))
+
+    __str__ = __repr__
+
+    def set_values(self, values=None):
+        self._algo_settings._set_categorical_values(self.name, values=values)
+
+    def enable_values(self, values, disable_others=False):
+        for value in values:
+            assert is_basestring(value)
+            self.set_values({value: {"enabled": True}})
+        if disable_others:
+            other_values = [key for key in self._algo_settings[self.name]["values"].keys() if key not in values]
+            for value in other_values:
+                self.set_values({value: {"enabled": False}})
+
+    def disable_values(self, values, enable_others=False):
+        for value in values:
+            assert is_basestring(value)
+            self.set_values({value: {"enabled": False}})
+        if enable_others:
+            other_values = [key for key in self._algo_settings[self.name]["values"].keys() if key not in values]
+            for value in other_values:
+                self.set_values({value: {"enabled": True}})
+
+
+class SingleValuedHyperparameterSettings(HyperparameterSettings):
+
+    def __init__(self, name, algo_settings, accepted_values=None):
+        super(SingleValuedHyperparameterSettings, self).__init__(name, algo_settings)
+        self.accepted_values = accepted_values
+
```

Comment:
Wouldn't it be relevant to add an accepted_type for the `SingleValuedHyperparameterSettings`, since they can be numerical or not (string etc)

Reply:
Yes totally -- already explicitly checked for `bool` but adding the type would be better.